src/pykeen/training/training_loop.py

```diff
@@ -701,7 +710,7 @@ def batch_size_search(
             batch_size = 8192
 
         # Set upper bound
-        batch_size = min(batch_size, self.triples_factory.num_triples)
+        batch_size = min(batch_size, self._num_triples)
```

Comment:
```suggestion
        batch_size = min(batch_size, triples_factory.num_triples)
```

If we pass a triples factory from the outside, it could be a different one than we gave in the constructor (although of course with the same label-to-id mapping).

Reply:
i like it. I just need to update the type annotations then